src/mame/drivers/plygonet.cpp

```diff
@@ -255,159 +271,164 @@ uint16_t polygonet_state::dsp56156_bootload_r()
 
    XXXX ---- ---- ----  . Reserved bits
    ---- ???- -?-- ----  . unknown
-   ---- ---- --x- ----  . [Bank Group A] Enable bit for "001c banking"?
-   ---- ---- ---x xx--  . [Group A bank control] Believed to bank memory from 0x8000-0xbfff
-   ---- ---- ---- --x-  . [Bank Group B] Enable bit for "0181 banking"?
-   ---- ---x x--- ---x  . [Group B bank control] Believed to bank various other memory regions
-
-   001c banking is fairly easy - it happens in a loop and writes from 8000 to bfff
-   0181 banking is very weird  - it happens in a nested loop and writes from 6000-6fff, 7000-7fff, and 8000-ffbf
-                                 bit 0002 turns on *just* before this happens.
+   ---- ---- --x- ----  . [Bank Group A] Enable bit
+   ---- ---- ---x xx--  . [Group A bank control] Banks memory from 0x8000-0xbfff
+   ---- ---- ---- --x-  . [Bank Group B] Enable bit
+   ---- ---x x--- ---x  . [Group B bank control] Banks memory at 0x6000-0x6fff and 0x7000-0x7fff (bits 7,8 only), and 0x8000-0xffbf (bits 7,8,0)
 */
 
-static uint8_t dsp56156_bank_group(device_t* cpu)
+void polygonet_state::dsp_portc_write(uint16_t data)
 {
-	uint16_t portC = downcast<dsp56156_device *>(cpu)->get_peripheral_memory(0xffe3);
-
-	/* If bank group B is on, it overrides bank group A */
-	if (portC & 0x0002)
-		return BANK_GROUP_B;
-	else if (portC & 0x0020)
-		return BANK_GROUP_A;
+	LOGMASKED(LOG_DSP_PORTC, "%s: DSP Port C write: %04x\n", machine().describe_context(), data);
+	m_dsp_portc = data;
 
-	return INVALID_BANK_GROUP;
-}
+	uint8_t bank_a_8_num = bitswap<3>(m_dsp_portc, 4, 3, 2);
+	m_dsp_bank_a_8->set_entry(bank_a_8_num);
 
-static uint8_t dsp56156_bank_num(device_t* cpu, uint8_t bank_group)
-{
-	uint16_t portC = downcast<dsp56156_device *>(cpu)->get_peripheral_memory(0xffe3);
+	uint8_t bank_b_67_num = bitswap<2>(m_dsp_portc, 7, 8);
+	m_dsp_bank_b_6->set_entry(bank_b_67_num);
+	m_dsp_bank_b_7->set_entry(bank_b_67_num);
```

Comment:
Trivial, but could you `const` the local bank numbers here, since they shouldn’t be changed after being assigned?  Just helps show intent.

Reply:
Just did a const-correctness pass over the whole file, should be better now.